pywal/export.py

```diff
@@ -47,11 +47,18 @@ def template(colors, input_file, output_file=None):
                     if func[0] != '.':
                         replace_str += "."
                     replace_str += "(".join(func) + ")"
+                else:
+                    # if it is an attribute i.e. rgb
+                    replace_str += '.' + fname
+                    new_color = function
+
+            if not isinstance(new_color, str):
+                new_color = new_color.strip
             # If the color was changed, replace with a unique identifier.
             if new_color is not colors[cname]:
-                template_data[i] = l.replace(
-                    replace_str, "color" + new_color.strip)
-                colors["color" + new_color.strip] = new_color
+                new_color_clean = new_color.replace('[', '_').replace(']', '_')
+                template_data[i] = l.replace(replace_str, "color" + new_color_clean)
```

Comment:
The CI tests complain about this line being too long, it should be shortened or split so that its less than 80 characters wide (including indentation).